boards/PDM/Inc/Io/Io_Efuse_22XS4200.h

```diff
@@ -0,0 +1,646 @@
+    PWM_Type_DC_8 = (1 << 7), //!< PWM Duty Cycle bit 7
+    PWM_Type_ON   = (1 << 8)  //!< PWM ON
+} PWM_Type_e;
+
+//! Enumeration for the channel configurations types
+typedef enum
+{
+    ChannelConfigType_Delay_0              = (1 << 0), //!< Switch-on Delay bit 0
+    ChannelConfigType_Delay_1              = (1 << 1), //!< Switch-on Delay bit 1
+    ChannelConfigType_Delay_2              = (1 << 2), //!< Switch-on Delay bit 2
+    ChannelConfigType_SlewRate_0           = (1 << 3), //!< Slew Rate bit 3
+    ChannelConfigType_SlewRate_1           = (1 << 4), //!< Slew Rate bit 4
+    ChannelConfigType_DirecControl_Disable = (1 << 5), //!< Direct Control Disable
+    ChannelConfigType_OpenLoad_InOffState  = (1 << 6), //!< Open Load Detection in Off-state
+    ChannelConfigType_OpenLoad_InOnState   = (1 << 7), //!< Open Load Detection in On-state
+    ChannelConfigType_ShortCircuitDetection_Disable = (1 << 8) //!< Short-circuit to Vpwr Detection Disable
+} ChannelConfigType_e;
+
+struct E_Fuse
+{
+    bool         watch_dog_state;
+    StatusType_e status;
+    struct ChannelFaults
+    {
+        FaultType_e channel_0_faults;
+        FaultType_e channel_1_faults;
+    } faults;
+    struct E_FuseChipSelect
+    {
+        GPIO_TypeDef *GPIO_Port;
+        uint16_t      GPIO_Pin;
+    } chip_select;
+    struct E_FuseCurrentSenseSync
+    {
+        GPIO_TypeDef *GPIO_Port;
+        uint16_t      GPIO_Pin;
+    } current_sense_sync;
+};
+
+/* Function Definitions */
+
+/**
+ * Initializes the SPI handle for the Efuses
+ * @param hspi Handle to the SPI peripheral used for the Efuses
+ */
+void Io_Efuse_22XS4200_Init(SPI_HandleTypeDef *const hspi);
+
+/**
+ * Configures the Efuse's registers using the values defined in: REGNAME_CONFIG
+ * @param *e_fuse Pointer to E_Fuse structure being configured
+ */
+void Io_Efuse_22XS4200_configureEfuse(struct E_Fuse *e_fuse);
+
+/**
+ * Sets current/temperature monitoring option of CSNS pin
+ * @param selection The desired monitoring function which can be selected by specifying the following values:
+ *                  CSNS_FUNCTION_DISABLED - Current sensing disabled
+ *                  CSNS_FUNCTION_CURRENT_CH0 - Current sensing for channel 0
+ *                  CSNS_FUNCTION_CURRENT_CH1 - Current sensing for channel 1
+ *                  CSNS_FUNCTION_TEMPERATURE - Ffuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed channels (for Parallel mode)
+ * @param *e_fuse Pointer to E_Fuse structure being configured
+ */
+void Io_Efuse_22XS4200_configure_channel_monitoring(uint8_t selection, struct E_Fuse *e_fuse);
+
+/**
+ * Gets the value of the Efuse's Status Register over SPI and updates the Efuse handle's operating status
+ * @param *e_fuse Pointer to E_Fuse structure being checked
+ */
+void Io_Efuse_22XS4200_UpdateStatus(struct E_Fuse *e_fuse);
+
+/**
+ * DGets the value of the Efuse's Fault Register over SPI and updates the Efuse handle's fault status
```

Comment:
Typo here, and another small nit; can we use `Get` instead of `Gets` for consistency? (Same apply to other jdocs in this file)
```suggestion
 * Get the value of the Efuse's Fault Register over SPI and updates the Efuse handle's fault status
```